### PR TITLE
Break out folders from status

### PR DESCRIPTION
**Pull request by ericnewcomer**
Preparation to allow deep linking to specific tickets.

* Uses new `temba-menu` component with an associated View
* Supports counts whenever we are ready for them and is collapsible
* Use cookies to save toggle state for details pane and menu collapse
* Adds url support when toggling folder or status, inits on reload appropriately

Fake counts:
Screen Shot 2021-07-21 at 2 56 52 PM

Collapsed:
Screen Shot 2021-07-21 at 2 55 26 PM

On staging.

### CODE REVIEW

File: temba/tickets/views.py [hunk 1]
```diff
@@ -81,23 +88,39 @@ class List(OrgPermsMixin, SmartListView):
...
         def get_queryset(self, **kwargs):
             return super().get_queryset(**kwargs).none()
 
+    class Menu(OrgPermsMixin, SmartTemplateView):
+        def render_to_response(self, context, **response_kwargs):
+            menu = [
+                {
+                    "id": folder.slug,
+                    "name": folder.name,
+                    "icon": folder.icon,
```
Comment (ericnewcomer) [inline temba/tickets/views.py:105]: If we add a `"count": 123` here it'll show up in the menu.

File: templates/tickets/ticket_list.haml [hunk 2]
```diff
@@ -151,13 +170,16 @@
 -block page-container
   .flex-col
     .flex.widget-container(style="height:100vh;margin-top:-121px;padding-top:121px;margin-bottom:-100px;padding-bottom:100px;")
-      .ticket-list.flex.flex-col.m-4.mr-2.pt-4
-        %temba-select(name="ticket-folder" onchange="handleFolderChanged(event)")
-          -for folder in folders
-            %temba-option(name='{{ folder.name }}' value="{{ folder.slug }}" icon="{{ folder.icon }}")
+      .folders.mt-4.ml-4.mb-4
+        %temba-menu(endpoint="{% url 'tickets.ticket_menu' %}" onchange="handleMenuChanged(event)" value="{{folder}}" collapsible)
```
Comment (ericnewcomer) [inline templates/tickets/ticket_list.haml:174]: `temba-menu` is based on list / options. Just give it an endpoint and it'll poll and update itself accordingly.

File: temba/tickets/views.py [hunk 3]
```diff
@@ -81,23 +81,39 @@ class List(OrgPermsMixin, SmartListView):
...
+            ]
+            return JsonResponse({"results": menu})
+
     class Folder(OrgPermsMixin, SmartListView):
         permission = "tickets.ticket_list"
 
         @classmethod
         def derive_url_pattern(cls, path, action):
-            return rf"^{path}/{action}/(?P<folder>{'|'.join(TicketFolder.all().keys())})/$"
+            return rf"^{path}/{action}/(?P<folder>{'|'.join(TicketFolder.all().keys())})/(?P<status>open|closed)/$"
```
Comment (rowanseymour) [inline temba/tickets/views.py:109]: I think if you make this..

```python
folders = '|'.join(TicketFolder.all().keys()
return rf"^ticket/(?P<folder>{folders})/(((?P<status>open|closed)/)?(?P<uuid>[a-z0-9\-]+)/)?$"
```

You don't need the 3 additions in urls.py

https://regex101.com/r/FbCzE1/1

Comment (ericnewcomer) [reply temba/tickets/views.py:109]: I was fighting with optionality, so went with the simpler option. Will take a look.

Comment (ericnewcomer) [reply temba/tickets/views.py:109]: FWIW, the pattern in question is actually for the main list page, not the folders. Had another go at optional parameters and with some tweaks arrived at something reasonable.

**Review by rowanseymour — APPROVED** [review]

**Pull request merged**